File: backend/app/models/schedule.py
from pydantic import BaseModel


class MeetingInput(BaseModel):
    day: str
    startTime: str
    endTime: str


class MeetingUpdateInput(BaseModel):
    originalMeeting: MeetingInput
    updatedMeeting: MeetingInput


class CourseInput(BaseModel):
    name: str
    doctor: str | None = None
    section: str | None = None
    building: str | None = None
    room: str | None = None
    meetings: list[MeetingInput]

File: frontend/src/types/schedule.ts
export type Day =
  | "sunday"
  | "monday"
  | "tuesday"
  | "wednesday"
  | "thursday"
  | "friday"
  | "saturday";

export type Meeting = {
  day: Day;
  startTime: string;
  endTime: string;
};

export type Course = {
  id: number;
  name: string;
  doctor?: string;
  building?: string;
  room?: string;
  section?: string;
  meetings: Meeting[];
};

export type NewCourse = Omit<Course, "id">;
